lib/dependency-manager-adapters/npm.js

```diff
@@ -36,14 +56,17 @@ module.exports = CoreObject.extend({
     debug('Write package.json with: \n', JSON.stringify(newPackageJSON));
 
     fs.writeFileSync(packageJSONFile, JSON.stringify(newPackageJSON, null, 2));
+    if (this.useYarn) {
+      this.noLock = true;
+    }
```

Comment:
Instead of a flag for `noLock`, how about passing it as an option to `_install()`

Installs when changing to a dependency set should not use the lockfile, the install on cleanup should, if present.